deflate.c

```diff
@@ -204,7 +204,7 @@ Z_INTERNAL deflate_allocs* alloc_deflate(PREFIX3(stream) *strm, int windowBits,
 
     /* Define sizes */
     int window_size = DEFLATE_ADJUST_WINDOW_SIZE((1 << windowBits) * 2);
-    int prev_size = (1 << windowBits) * sizeof(Pos);
+    int prev_size = ((unsigned long long)1 << windowBits) * sizeof(Pos);
```

Comment:
`UINT64_C(1)`... Cast is never used with constant literals. Bit width of left side must match bit width of right side when bit-shifting, otherwise VC++ will issue warning.

Reply:
This is the only way I could get the warning to go away..
```c
int prev_size = (1 << windowBits) * (int)sizeof(Pos);
```

Reply:
> This is the only way I could get the warning to go away..
> 
> ```c
> int prev_size = (1 << windowBits) * (int)sizeof(Pos);
> ```

That might actually be the cleanest way as it avoids the need for truncating the whole expression to fit in `int`... `sizeof́()` returning `size_t` has been causing a lot of issues since 64-bit processors were introduced.

Reply:
Addressed. I squashed the commits too.